src/main/java/org/generationcp/middleware/service/impl/study/advance/AdvanceServiceImpl.java

```diff
@@ -153,15 +165,34 @@ public AdvanceServiceImpl(final HibernateSessionProvider sessionProvider) {
 
 	@Override
 	public List<Integer> advanceStudy(final Integer studyId, final AdvanceStudyRequest request) {
-		return this.advance(studyId, request);
+		return this.advance(studyId, request, false, null, false);
+	}
+
+	@Override
+	public List<AdvanceGermplasmPreview> advanceStudyPreview(final Integer studyId, final AdvanceStudyRequest request) {
+		final List<AdvanceGermplasmPreview> advanceGermplasmPreviewList = new ArrayList<>();
+		this.advance(studyId, request, true, advanceGermplasmPreviewList, false);
+		Collections.sort(advanceGermplasmPreviewList, Comparator.comparing(AdvanceGermplasmPreview::getEntryNumber));
+
+		return advanceGermplasmPreviewList;
 	}
 
 	@Override
 	public List<Integer> advanceSamples(final Integer studyId, final AdvanceSamplesRequest request) {
-		return this.advance(studyId, request);
+		return this.advance(studyId, request, false, null, true);
+	}
+
+	@Override
+	public List<AdvanceGermplasmPreview> advanceSamplesPreview(final Integer studyId, final AdvanceSamplesRequest request) {
+		final List<AdvanceGermplasmPreview> advanceGermplasmPreviewList = new ArrayList<>();
+		this.advance(studyId, request, true, advanceGermplasmPreviewList, true);
+		Collections.sort(advanceGermplasmPreviewList, Comparator.comparing(AdvanceGermplasmPreview::getEntryNumber));
+
+		return advanceGermplasmPreviewList;
 	}
 
-	private List<Integer> advance(final Integer studyId, final AdvanceRequest request) {
+	private List<Integer> advance(final Integer studyId, final AdvanceRequest request,
+		final boolean isPreview, final List<AdvanceGermplasmPreview> advanceGermplasmPreviewList, final boolean isSample) {
```

Comment:
Instead of adding isSample parameter, maybe we can create a request visitor (AdvanceRequestVisitor) class to check if a given id is excluded, depending if the request is AdvanceStudyRequest or AdvanceSampleRequest.

Also create a new visitor get the uniqueId for preview.